server/matriz.js

```diff
@@ -52,37 +62,58 @@ function rmvPosition(position) {
 function verify(position, symb) {
   let x = position[0]
   let y = position[1]
+  console.log(symb)
 
-  loop1: for (let i = 0; i < matriz.length; i++) {
-    loop2: for (let j = 0; j < matriz.length; j++) {
+  for (let i = 0; i < matriz.length; i++) {
+    for (let j = 0; j < matriz.length; j++) {
       // horizontal check
       if (x == i && matriz[x][0] == matriz[x][1] && matriz[x][1] == matriz[x][2]) {
-        return 2
+        answer.array = [[x, 0], [x, 1], [x, 2]]
+        answer.type = 2
+        return answer
       }
 
       // Vertical check
       if (y == j && matriz[0][y] == matriz[1][y] && matriz[1][y] == matriz[2][y]) {
-        return 2
+        answer.array = [[0, y], [1, y], [2, y]]
+        answer.type = 2
+        return answer
       }
 
       // Main Diagonal check
       if (x == y && matriz[0][0] == matriz[1][1] && matriz[1][1] == matriz[2][2]) {
-        return 2
+        answer.array = [[0, 0], [1, 1], [2, 2]]
+        answer.type = 2
+        return answer
       }
 
       // Secondary Diagonal check
       if (
-        (x == y || (x == 0 && y == 2) || (x == 2 && y == 0)) &&
+        ((x == 1 && y == 1) || (x == 0 && y == 2) || (x == 2 && y == 0)) &&
         matriz[0][2] == matriz[1][1] &&
         matriz[1][1] == matriz[2][0]
       ) {
-        return 2
+        answer.array = [[0, 2], [1, 1], [2, 0]]
+        answer.type = 2
+        return answer
       }
     }
   }
-  return 1
+  count++
+  // Tie check
+  if (count == 9) {
+    answer.array = [x, y]
```

Comment:
Talvez eu esteja errado, mas me parece que aqui vc seta `answer.array = [x, y]` (na linha 105), mas `answer.array` só tem valor caindo dentro desse if (tie) ou dentro de alguns if do `for` (win). E no caso de uma jogada normal?

Reply:
Resolvido